pkg/3scale/amp/operator/redis_reconciler.go

```diff
@@ -124,3 +147,68 @@ func Redis(apimanager *appsv1alpha1.APIManager, client client.Client) (*componen
 	}
 	return component.NewRedis(opts), nil
 }
+
+func (r *RedisReconciler) redisConfigCmMutator(existingObj, desiredObj common.KubernetesObject) (bool, error) {
+	existing, ok := existingObj.(*corev1.ConfigMap)
+	if !ok {
+		return false, fmt.Errorf("%T is not a *v1.ConfigMap", existingObj)
+	}
+	desired, ok := desiredObj.(*corev1.ConfigMap)
+	if !ok {
+		return false, fmt.Errorf("%T is not a *v1.ConfigMap", desiredObj)
+	}
+
+	update := false
+	fieldUpdated := reconcilers.ConfigMapReconcileField(desired, existing, "redis.conf")
+	update = update || fieldUpdated
+
+	return update, nil
+}
+
+func (r *RedisReconciler) isInternalRedis() (bool, error) {
+	if r.apiManager.Spec.ExternalComponents != nil &&
```

Comment:
apim has already a function "IsExternal" that could be used here I think

Reply:
Yes, apim has IsExternal  function, but it contains the code related to HighAvailability, that seems not so related to current case, so I like more current implementation. If it's not critical for you. Thanks Michal

Reply:
I recommend doing something like this instead:
```
backendRedisExternal false
systemRedisExternal false

if apimInstance.IsExternal(appsv1alpha1.BackendRedis) {
    backendRedisExternal = true
}
if apimInstance.IsExternal(appsv1alpha1.SystemRedis) {
    systemRedisExternal = true
}

return !backendRedisExternal || !systemRedisExternal

I think this should work.
In general, it's ok to do it the way you have it but it can be simplified. 

I would use what we have already.

Reply:
Thank you for suggestion. Still seems to me, my implementation is not dependent to high-availability (mapHighAvailabilityToExternalComponents()), so maybe we can use it as is. Thanks

Reply:
OK, that's fine. Just fyi, this is not related to high availability, it's only a part of the isExternall function to support old implementation of external db's where they were specified in HA section...